src/c/devmap.c

```diff
@@ -99,7 +100,11 @@ static void add_locked (edgex_devmap_t *map, const edgex_device *newdev)
     edgex_map_set (&map->profiles, dup->profile->name, dup->profile);
   }
   edgex_map_set (&map->devices, dup->name, dup);
-  edgex_device_autoevent_start (map->svc, dup);
+  result = search_devsdk_device(device_first, dup->name);
+  if (result)
+  {
+    edgex_device_autoevent_start (map->svc, dup);
```

Comment:
Not sure what effect this is intended to have - looks as though autoevents will only be started for devices added via edgex_add_device?

Reply:
> Not sure what effect this is intended to have - looks as though autoevents will only be started for devices added via edgex_add_device?

@iain-anderson Thanks for the review... I am putting my observation and analysis below point wise:

**The autoevents gets invoked in 2 ways for 2 different scenarios:**

**1. When we boot target for the very first time:**
In this scenario the device profile and default device interfaces (viz. eth0 in this case) gets added to metadata for the very first time.
After the successful addition of the device the autoevents gets scheduled and invoked from the API's in the below order:

edgex_devmap_populate_devices---->add_locked---->edgex_device_autoevent_start---->ae_runner


**2. When we reboot the target or poweroff and start the target:**
In this scenario the device profile and default device interfaces (viz. eth0 in this case) is already in the entry and hence when sdk tries
to add the device in metadata, it goes for a check initially where it is found that device already exits in the entry and hence the device 
profile and device just needs to be updated in the sdk in order to invoke autoevents with the help of device profile update callback.

**Issue in target machine:**
It runs fine when target gets booted for the very first time. But after reboot a flaky issue is found. Only 1st autoevent is invoked and rest 
all of them are not appearing. Hence, in this case when we try to acquire value of rest of the autoevent resources from edgex developer's UI
we get 502 error.

**Root cause of the issue:** 
After reboot when service starts, sometimes the below order of API's gets triggered first before the device exist check while device addition.  

edgex_devmap_populate_devices---->add_locked--->edgex_device_autoevent_start---->ae_runner

Since the device is already in metadata even before the existing check, the above API order tries to invoke the autoevents for the device with schedulers.
The ae_runner works in schedulers running parallely. Meanwhile there is an attempt for adding device where it is found under the check that
device already exists. Under the same check there is an API edgex_device_release call which has call for edgex_device_autoevent_stop inside
which actually stops and deletes the remaining autoevent schedulers which are yet to be run. Once stopped further no autoevents gets triggered ever.
In this scenario no device profile update callback gets invoked which also triggers the autoevents. 

**Possible fix:**
We can make a data structure which would contain the list of only newly added device. We will do a search operation on the data structure just before edgex_device_autoevent_start 
during the API order call as mentioned above earlier. This will make sure that autoevents gets invoked with this API order call only for newly added device. 

edgex_devmap_populate_devices---->add_locked---->search_devsdk_device--->{if search for a new device successful}---->edgex_device_autoevent_start---->ae_runner

For devices already in the metadata entry, the autoevents will be automatically scheduled and invoked by the device profile update callback.